### DISCUSSION TITLE
How can I get information of scala attributes?

### DISCUSSION THREAD

**Discussion by ehdrb01** [scala/scala3#22244] [General Question] [answered]
Hello!

I want to know the types of attributes in the scala language. I understand scala adds some attributes into class file which is a result of compilation process.

<img width="756" alt="Screenshot 2024-12-19 at 6 02 02 PM" src="https://github.com/user-attachments/assets/659f4d8d-3e07-4396-b411-28c181b1c258" />

This is a JVM specification and it shows all attribute types from the JVM. However, I cannot find scala specific attributes. Can I get your help?

Thanks!

**Comment by bishabosha** [comment, marked as answer]
This parser should tell you what you need: https://github.com/scalacenter/tasty-query/blob/main/tasty-query/shared/src/main/scala/tastyquery/reader/classfiles/ClassfileParser.scala

I dont know if there is a spec published of class file additions

**Reply by ehdrb01** [reply]
Thank you for your response! I will check it!
